packages/react-dom/src/__tests__/ReactDOMServerLifecycles-test.internal.js

```diff
@@ -40,4 +40,30 @@ describe('ReactDOMServerLifecycles', () => {
     // De-duped
     ReactDOMServer.renderToString(<Component />);
   });
+
+  describe('react-lifecycles-compat', () => {
+    // TODO Replace this with react-lifecycles-compat once it's been published
+    function polyfill(Component) {
+      Component.prototype.componentWillMount = function() {};
+      Component.prototype.componentWillMount.__suppressDeprecationWarning = true;
+      Component.prototype.componentWillReceiveProps = function() {};
+      Component.prototype.componentWillReceiveProps.__suppressDeprecationWarning = true;
```

Comment:
This pollutes the `react` module but there's no `resetModules()` in `afterEach`. Let's add it so that any tests added later to this file aren't affected by this monkeypatching.

Reply:
Nice call.